kyuubi-server/src/test/scala/org/apache/kyuubi/operation/KyuubiOperationPerUserSuite.scala

```diff
@@ -153,4 +154,44 @@ class KyuubiOperationPerUserSuite extends WithKyuubiServer with SparkQueryTests
       }
     }
   }
+
+  test("support to interrupt the thrift request if remote engine is broken") {
+    withSessionConf(Map(
+      KyuubiConf.ENGINE_ALIVE_PROBE_ENABLED.key -> "true",
+      KyuubiConf.ENGINE_ALIVE_PROBE_INTERVAL.key -> "1000",
+      KyuubiConf.ENGINE_ALIVE_TIMEOUT.key -> "3000",
+      KyuubiConf.OPERATION_THRIFT_CLIENT_REQUEST_MAX_ATTEMPTS.key -> "10000"))(Map.empty)(
+      Map.empty) {
+      withSessionHandle { (client, handle) =>
+        val preReq = new TExecuteStatementReq()
+        preReq.setStatement("select engine_name()")
+        preReq.setSessionHandle(handle)
+        preReq.setRunAsync(false)
+        client.ExecuteStatement(preReq)
+
+        new Thread("async-exit") {
+          override def run(): Unit = {
+            withSessionHandle { (client2, handle2) =>
+              Thread.sleep(2000)
```

Comment:
Instead of `Thread.sleep(2000) / java.lang.Thread', 'sleep', 3600000l)` with flakiness, I guess we can find a better way

Reply:
how about now?